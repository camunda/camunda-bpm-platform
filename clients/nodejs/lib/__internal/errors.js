/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. See the NOTICE file
 * distributed with this work for additional information regarding copyright
 * ownership. Camunda licenses this file to you under the Apache License,
 * Version 2.0; you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

// Client
const MISSING_BASE_URL =
  "Couldn't instantiate Client, missing configuration parameter 'baseUrl'";
const WRONG_INTERCEPTOR =
  "Interceptors should be a function or an array of functions";
const WRONG_MIDDLEWARES =
  "Middleware(s) should be a function or an array of functions";
const ALREADY_REGISTERED = "Subscription failed, already subscribed to topic";
const MISSING_HANDLER = "Subscription failed, missing handler function";

// Task Service
const MISSING_TASK = "Couldn't complete task, task id is missing";
const MISSING_ERROR_CODE =
  "Couldn't throw BPMN Error, no error code was provided";
const MISSING_DURATION = "Couldn't lock task, no duration was provided";
const MISSING_NEW_DURATION =
  "Couldn't extend lock time, no new duration was provided";

// Basic Auth Interceptor
const MISSING_BASIC_AUTH_PARAMS =
  "Couldn't instantiate BasicAuthInterceptor, missing configuration parameter " +
  "'username' or 'password'";

// Keycloak Auth Interceptor
const MISSING_KEYCLOAK_AUTH_PARAMS =
  "Couldn't instantiate KeycloakAuthInterceptor, missing configuration parameter " +
  "'tokenEndpoint', 'clientId' or 'clientSecret'";
const UNEXPECTED_KEYCLOAK_TOKEN_RESULT =
  "Couldn't get access token from Keycloak provider; got";

// FileService
const MISSING_FILE_OPTIONS =
  "Couldn't create a File, make sure to provide one of the following" +
  " parameters: \n- path \ntypedValue";

const WRONG_SORTING =
  "Couldn't instantiate Client, 'sorting' parameter should be an array.";
const WRONG_SORTING_SORT_BY =
  "Couldn't instantiate Client, wrong 'sorting.sortBy' parameter. Possible values: ";
const WRONG_SORTING_SORT_ORDER =
  "Couldn't instantiate Client, wrong 'sorting.sortOrder' parameter. Possible values: ";

export {
  MISSING_BASE_URL,
  ALREADY_REGISTERED,
  MISSING_HANDLER,
  MISSING_TASK,
  WRONG_INTERCEPTOR,
  MISSING_ERROR_CODE,
  MISSING_DURATION,
  MISSING_NEW_DURATION,
  MISSING_BASIC_AUTH_PARAMS,
  MISSING_KEYCLOAK_AUTH_PARAMS,
  UNEXPECTED_KEYCLOAK_TOKEN_RESULT,
  WRONG_MIDDLEWARES,
  MISSING_FILE_OPTIONS,
  WRONG_SORTING,
  WRONG_SORTING_SORT_BY,
  WRONG_SORTING_SORT_ORDER,
};
