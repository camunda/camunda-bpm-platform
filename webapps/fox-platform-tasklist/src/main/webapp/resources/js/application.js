// NOTICE!! DO NOT USE ANY OF THIS JAVASCRIPT
// IT'S ALL JUST JUNK FOR OUR DOCS!
// ++++++++++++++++++++++++++++++++++++++++++

!function ($) {

  $(function(){

    // Disable certain links in docs
    $('section [href^=#]').click(function (e) {
      e.preventDefault()
    })

    // make code pretty
    window.prettyPrint && prettyPrint()

    // add-ons
    $('.add-on :checkbox').on('click', function () {
      var $this = $(this)
        , method = $this.attr('checked') ? 'addClass' : 'removeClass'
      $(this).parents('.add-on')[method]('active')
    })

    // position static twipsies for components page
    if ($(".twipsies a").length) {
      $(window).on('load resize', function () {
        $(".twipsies a").each(function () {
          $(this)
            .tooltip({
              placement: $(this).attr('title')
            , trigger: 'manual'
            })
            .tooltip('show')
          })
      })
    }

    // add tipsies to grid for scaffolding
    if ($('#grid-system').length) {
      $('#grid-system').tooltip({
          selector: '.show-grid > div'
        , title: function () { return $(this).width() + 'px' }
      })
    }

    // fix sub nav on scroll
    var $win = $(window)
      , $nav = $('.subnav')
      , navTop = $('.subnav').length && $('.subnav').offset().top - 40
      , isFixed = 0

    processScroll()

    // hack sad times - holdover until rewrite for 2.1
    $nav.on('click', function () {
      if (!isFixed) setTimeout(function () {  $win.scrollTop($win.scrollTop() - 47) }, 10)
    })

    $win.on('scroll', processScroll)

    function processScroll() {
      var i, scrollTop = $win.scrollTop()
      if (scrollTop >= navTop && !isFixed) {
        isFixed = 1
        $nav.addClass('subnav-fixed')
      } else if (scrollTop <= navTop && isFixed) {
        isFixed = 0
        $nav.removeClass('subnav-fixed')
      }
    }

    // tooltip demo
    $('.tooltip-demo.well').tooltip({
      selector: "a[rel=tooltip]"
    })

    $('.tooltip-test').tooltip()
    $('.popover-test').popover()

    // popover demo
    $("a[rel=popover]")
      .popover()
      .click(function(e) {
        e.preventDefault()
      })

    // button state demo
    $('#fat-btn')
      .click(function () {
        var btn = $(this)
        btn.button('loading')
        setTimeout(function () {
          btn.button('reset')
        }, 3000)
      })

    // carousel demo
    $('#myCarousel').carousel()

    // javascript build logic
    var inputsComponent = $("#components.download input")
      , inputsPlugin = $("#plugins.download input")
      , inputsVariables = $("#variables.download input")

    // toggle all plugin checkboxes
    $('#components.download .toggle-all').on('click', function (e) {
      e.preventDefault()
      inputsComponent.attr('checked', !inputsComponent.is(':checked'))
    })

    $('#plugins.download .toggle-all').on('click', function (e) {
      e.preventDefault()
      inputsPlugin.attr('checked', !inputsPlugin.is(':checked'))
    })

    $('#variables.download .toggle-all').on('click', function (e) {
      e.preventDefault()
      inputsVariables.val('')
    })

    // request built javascript
    $('.download-btn').on('click', function () {

      var css = $("#components.download input:checked")
            .map(function () { return this.value })
            .toArray()
        , js = $("#plugins.download input:checked")
            .map(function () { return this.value })
            .toArray()
        , vars = {}
        , img = ['glyphicons-halflings.png', 'glyphicons-halflings-white.png']

    $("#variables.download input")
      .each(function () {
        $(this).val() && (vars[ $(this).prev().text() ] = $(this).val())
      })

      $.ajax({
        type: 'POST'
      , url: /\?dev/.test(window.location) ? 'http://localhost:3000' : 'http://bootstrap.herokuapp.com'
      , dataType: 'jsonpi'
      , params: {
          js: js
        , css: css
        , vars: vars
        , img: img
      }
      })
    })
  })

// Modified from the original jsonpi https://github.com/benvinegar/jquery-jsonpi
$.ajaxTransport('jsonpi', function(opts, originalOptions, jqXHR) {
  var url = opts.url;

  return {
    send: function(_, completeCallback) {
      var name = 'jQuery_iframe_' + jQuery.now()
        , iframe, form

      iframe = $('<iframe>')
        .attr('name', name)
        .appendTo('head')

      form = $('<form>')
        .attr('method', opts.type) // GET or POST
        .attr('action', url)
        .attr('target', name)

      $.each(opts.params, function(k, v) {

        $('<input>')
          .attr('type', 'hidden')
          .attr('name', k)
          .attr('value', typeof v == 'string' ? v : JSON.stringify(v))
          .appendTo(form)
      })

      form.appendTo('body').submit()
    }
  }
})

}(window.jQuery)