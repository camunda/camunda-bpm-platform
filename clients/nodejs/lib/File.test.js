/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. See the NOTICE file
 * distributed with this work for additional information regarding copyright
 * ownership. Camunda licenses this file to you under the Apache License,
 * Version 2.0; you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { jest } from "@jest/globals";
import { File } from "../index.js";
import { MISSING_FILE_OPTIONS } from "./__internal/errors.js";

describe("File", () => {
  describe("constructor", () => {
    it("should throw Error when neither localPath nor typedValue are provided", () => {
      expect(() => new File()).toThrowError(MISSING_FILE_OPTIONS);
    });

    it("should create instance with proper values when typedValue is provided", () => {
      // given
      const typedValue = {
        valueInfo: {
          filename: "somefile",
          mimetype: "application/json",
          encoding: "utf-8",
        },
      };
      const file = new File({ localPath: "foo", typedValue });

      // then
      expect(file).toMatchSnapshot();
    });

    it("should create instance with proper values", () => {
      // given
      const options = {
        filename: "somefile",
        mimetype: "application/json",
        encoding: "utf-8",
        localPath: "foo",
      };

      const file = new File(options);

      // then
      expect(file).toMatchSnapshot();
    });
  });

  describe("load", () => {
    it("should load content from remotePath when it's provided", async () => {
      // given
      const engineService = {
        get: jest
          .fn()
          .mockImplementation(() => Promise.resolve(Buffer.from("", "utf-8"))),
      };
      const remotePath = "some/remote/path";
      const expectedBuffer = Buffer.from(await engineService.get(remotePath));
      const file = await new File({ remotePath, engineService }).load();

      // then
      expect(file.content).toEqual(expectedBuffer);
    });

    it("should load content from localPath when it's provided", async () => {
      // given
      const localPath = "some/local/path";
      let file = await new File({ localPath });
      file.__readFile = jest.fn().mockImplementation(() => {
        return Promise.resolve("some content");
      });
      const expectedContent = "some content";
      file = await file.load();
      const content = file.content;

      // then
      expect(content).toBe(expectedContent);
    });
  });

  describe("createTypedValue", () => {
    it("should create typedValue with provided parameters", async () => {
      // given
      const valueInfo = {
        filename: "somefile",
        mimetype: "application/text",
        encoding: "utf-8",
      };
      const value = "this some random value";
      const expectedTypedValue = {
        value: Buffer.from(value).toString("utf-8"),
        type: "file",
        valueInfo,
      };
      const engineService = {
        get: jest.fn().mockImplementation(() => Promise.resolve(value)),
      };
      const remotePath = "some/remote/path";
      const file = await new File({
        remotePath,
        engineService,
        ...valueInfo,
      }).load();

      // then
      expect(file.createTypedValue()).toEqual(expectedTypedValue);
    });
  });
});
