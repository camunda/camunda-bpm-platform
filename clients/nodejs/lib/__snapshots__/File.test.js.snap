// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File constructor should create instance with proper values 1`] = `
File {
  "__readFile": [Function],
  "content": "",
  "createTypedValue": [Function],
  "encoding": "utf-8",
  "filename": "somefile",
  "load": [Function],
  "localPath": "foo",
  "mimetype": "application/json",
}
`;

exports[`File constructor should create instance with proper values when typedValue is provided 1`] = `
File {
  "__readFile": [Function],
  "content": "",
  "createTypedValue": [Function],
  "encoding": "utf-8",
  "filename": "somefile",
  "load": [Function],
  "localPath": "foo",
  "mimetype": "application/json",
}
`;
